microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,156 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
 
     return [u for u in uncovered if (u['sequence'] != '') &
                                     (len(u['sequence']) >= min_subseq_len)]
+
+
+def get_q_id(hit):
+    """ Returns the query ID for a hit.
+
+    Parameters
+    ----------
+    A hit parsed from an HHsearch output file, i.e. dict with at least the key
```

Comment:
The design of the data structure is governed by the information in the file that is parsed, which is the output of an HHsearch run. A hit has some score information like probability, e- and p- value and others and the block of the alignment of query sequence vs. hmm consensus sequence. The lines of the alignment come with these strange names 'Q Consensus', ...

Since this function is a highly specialized one, I'd be happy if I could leave it as is. :-/ Or do you have a suggestion for a better text?